### PR TITLE
Remove unreachable code

### LINKED ISSUE #12: Write tests for not yet covered functionality
There are still parts not covered by tests. Coverage results can be found here: https://app.codecov.io/gh/flexmock/flexmock so let's have everything covered.

### PR DESCRIPTION
**Pull request by ollipa**
Removed unreachable code to reach 100% test coverage.

Closes #12

### CODE REVIEW

File: src/flexmock/_api.py [hunk 1]
```diff
@@ -1315,25 +1315,20 @@ def get_flexmock_expectation(
...
+            if expectation._name == name and expectation._match_args(args):
+                if expectation in cls.ordered or not expectation._ordered and not found:
+                    found = expectation
+        if found and found._ordered:
+            cls._verify_call_order(found, args)
+        return found
 
     @classmethod
     def _verify_call_order(cls, expectation: Expectation, args: Dict[str, Any]) -> None:
-        if not cls.ordered:
```
Comment (christophe-riolo) [inline src/flexmock/_api.py:1331]: Why can't it be empty ? I guess it's because we never instantiate directly ?

File: src/flexmock/_api.py [hunk 2]
```diff
@@ -1306,7 +1306,7 @@ def reset(cls) -> None:
 
     @classmethod
     def get_flexmock_expectation(
-        cls, obj: Mock, name: Optional[str] = None, args: Optional[Any] = None
```
Comment (christophe-riolo) [inline src/flexmock/_api.py:1309]: Thanks type checking :relieved:

File: src/flexmock/_api.py [hunk 3]
```diff
@@ -1315,25 +1315,20 @@ def get_flexmock_expectation(
             args = {"kargs": args, "kwargs": {}}
         if not isinstance(args["kargs"], tuple):
             args["kargs"] = (args["kargs"],)
-        if name and obj in cls.flexmock_objects:
```
Comment (christophe-riolo) [inline src/flexmock/_api.py:1318]: I guess we can have `obj not it cls.flexmock_objects` if that classmothod is called directly by the user; should it become a private method ?

Comment (ollipa) [reply src/flexmock/_api.py:1318]: `FlexmockContainer` itself is behind private module `_api` so I don't think we need to mark these methods private.

File: src/flexmock/_api.py (hunk 1 above)
Comment (ollipa) [reply src/flexmock/_api.py:1331]: Yeah this method is only called from `get_flexmock_expectation` and that already checks that the expectation is ordered, so it should not be possible for `cls.ordered` to be empty. From the public API, this code is unreachable.

**Review by christophe-riolo — APPROVED** [review]
LGTM :+1:

**Pull request merged**